java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -370,37 +370,39 @@ private API getXPAPI(ExpressionTree et, VisitorState state) {
 
   private boolean methodRecordMatcher(
       MethodRecord methodRecord, VisitorState state, MethodInvocationTree mit) {
+
     // Method name must match methodRecord.getMethodName()
     boolean nameMatches =
         (methodRecord.isStatic() ? staticMethod().anyClass() : instanceMethod().anyClass())
             .named(methodRecord.getMethodName())
             .matches(mit, state);
-    if (nameMatches) {
-      // Method's receiver must match record's receiver type (if any)
-      boolean receiverTypeMatches =
-          !methodRecord.getReceiverType().isPresent()
-              || receiverOfInvocation(
-                      (receiver, st) ->
-                          isSameType(methodRecord.getReceiverType().get()).matches(receiver, st))
-                  .matches(mit, state);
-      if (receiverTypeMatches) {
-        // Method must have flag at argument index specified by record (if any)
-        boolean argumentMatchesFlagName =
-            !methodRecord.getArgumentIdx().isPresent()
-                || argument(
-                        methodRecord.getArgumentIdx().get(),
-                        (arg, st) -> isArgumentMatchesFlagName(arg, state))
-                    .matches(mit, state);
-        if (argumentMatchesFlagName) {
-          // Method's return must match record's return type (if any)
-          return methodRecord
-              .getReturnType()
-              .map(typeString -> isSameType(typeString).matches(mit, state))
-              .orElse(true);
-        }
-      }
-    }
-    return false;
+    if (!nameMatches) return false;
+
+    // Method's receiver must match record's receiver type (if any)
+    boolean receiverTypeMatches =
+        !methodRecord.getReceiverType().isPresent()
+            || receiverOfInvocation(
+                    (receiver, st) ->
+                        isSameType(methodRecord.getReceiverType().get()).matches(receiver, st))
+                .matches(mit, state);
+    if (!receiverTypeMatches) return false;
```

Comment:
I believe our coding guidelines recommend all conditionals to have a block for their branches (see [this](http://errorprone.info/bugpattern/MissingBraces) check, which we use internally and on NullAway, and which might make sense for this project too). So, if we are following that same style for Piranha, this probably should be:

```
if (!receiverTypeMatches) {
   return false;
}
```